client/components/preview.tsx

```diff
@@ -41,6 +41,36 @@ const Preview: NextPage<{ expense: ExpenseModel }> = (props) => {
     }
   });
 
+  function truncate(str: string, n: number) {
+    return (str.length > n) ? str.substring(0, n - 1) + '...' : str;
+  };
+
+  function formatTimeType(value: number, type: string): string {
+    if (value == 0) return '';
+    if (Math.abs(value) == 1) return `  • ${Math.abs(value)} ${type}`;
+    return `${Math.abs(value)} ${type}s`;
+  }
+
+  function calculateTimeDiff(expenseDue: Date): string {
```

Comment:
A quite cool function that makes paymentDue to "X days" and more depending on time left.